azm/amd64/assembler.h

```diff
@@ -0,0 +1,83 @@
+// SPDX-FileCopyrightText: 2023 Robin Lindén <dev@robinlinden.eu>
+//
+// SPDX-License-Identifier: BSD-2-Clause
+
+#ifndef AZM_AMD64_ASSEMBLER_H_
+#define AZM_AMD64_ASSEMBLER_H_
+
+#include <cstdint>
+#include <iostream>
+#include <optional>
+#include <utility>
+#include <vector>
+
+namespace azm::amd64 {
+
+enum class Register {
```

Comment:
Might make sense to call this enum `Register32` or something. That way you can implement a mov overload for, e.g, a 64-bit immediate to register, and you won't have to do a check to see if the destination is valid, because you'll be using a `Register64` there.

Reply:
Yeah, that's a good point, and the same reason there's an `Imm32` and not an `Imm`. :P Thank you!